Constants/ColorConstant.swift

```diff
@@ -0,0 +1,29 @@
+//
+//  ColorConstant.swift
+//  TimePad
+//
+//  Created by Emre Alpago on 10.09.2022.
+//
+
+import Foundation
+import UIKit
+
+class ColorConstant {
+    static let shared = ColorConstant()
```

Comment:
bu bir singleton ise init methodunun dışarıya kapalı olması lazım. private init yapıp sadece shared'ın kullanılması için. Class'ı da kimse inherit etmiyorsa final yapmak iyi olur. 

Singleton design Pattern'ı burada kullanmak nasıl bir + sağladı sence? Genelde singleton, adından da anlaşıldığı üzere tekil dataya ihtiyaç duyulduğunda kullanılıyor. Yani tüm app boyunca data'nın sadece 1 yerde ve aynı olması istendiği durum. Mesela örnek veriyorum tüm app boyunca bir user objen var ve bunu store etmen lazım runtime'da bunu singletonda tutup tüm app'in o user'a kolayca erişmesini sağlayabilirsin. Ama renkler zaten hardcoded. Singleton'a ihtiyaç yok. 


burada enum kullanmak daha kolay erişmeyi sağlayabilir
bu yazıya bakabilirsin. Renkler için de color assets kullanmanı öneririm. 


https://medium.com/developerinsider/xcode-pro-tips-how-to-use-asset-catalogs-to-support-named-colors-8959ab51fd84
https://medium.com/teamarimac/enumerate-like-a-pro-6c8fdac9066b